src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,92 +48,1243 @@
+            Collections.sort(teamMembersOfTeam);
+            
+            if (isGiver) {
+                addMissingParticipantsForTeamToSectionPanelWithModerationButton(sectionPanel, teamWithoutResponses, 
+                                                                                teamMembersOfTeam, isGiver);
+            } else {
+                addMissingParticipantsForTeamToSectionPanelWithoutModerationButton(sectionPanel, teamWithoutResponses, 
+                                                teamMembersOfTeam, isGiver);
+            }
+        }
+    }
+
+    private String getCurrentTeam(FeedbackSessionResultsBundle bundle, String giverIdentifier) {
+        String currentTeam;
+        if (bundle.isParticipantIdentifierInstructor(giverIdentifier)) {
+            currentTeam = Const.USER_TEAM_FOR_INSTRUCTOR;
+        } else {
+            currentTeam = bundle.getTeamNameForEmail(giverIdentifier);
+            if (currentTeam.equals("")) {
+                currentTeam = bundle.getNameForEmail(giverIdentifier);
+            }
+        }
+        return currentTeam;
+    }
+    
+    private String getCurrentGiverSection(
+                                    Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesFromGiver) {
+        return getCurrentSection(responsesFromGiver, true);
+    }
+    
+    private String getCurrentRecipientSection(
+                                    Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesFromRecipient) {
+        return getCurrentSection(responsesFromRecipient, false);
+    }
+    
+
+    /**
+     * Uses the first response to get the current section
+     * @param responses
+     */
+    private String getCurrentSection(
+                        Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responses,
+                        boolean isGiver) {
+        String currentSection = Const.DEFAULT_SECTION;
+        // update current section
+        // retrieve section from the first response of this user
+        // TODO simplify by introducing more data structures into bundle
+        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> responsesFromGiverForQuestion : 
+                                                                                     responses.getValue().entrySet()) {
+            if (responsesFromGiverForQuestion.getValue().isEmpty()) {
+                continue;
+            }
+            FeedbackResponseAttributes firstResponse = responsesFromGiverForQuestion.getValue().get(0);
+            currentSection = isGiver ? firstResponse.giverSection : firstResponse.recipientSection ;
+            break;
+        }
+        
+        return currentSection;
+    }
+
+    private void createMissingParticipantPanelsWithModerationButtonForPrevTeam(
```

Comment:
Does the `ForPrevTeam` part in its name have any significance?
Will there be more methods, `ForCurrentTeam`, `ForNextTeam`, and so on?
Or is it just a team in general?

(If it's just a general team, do rename the variable as well.)